src/Core/Resolvers/MultipleCreateOrderHelper.cs

```diff
@@ -61,6 +61,17 @@ public class MultipleCreateOrderHelper
                     subStatusCode: DataApiBuilderException.SubStatusCodes.EntityNotFound);
             }
 
+            DatabaseTable sourceDbTable = (DatabaseTable)sourceDbObject;
+            DatabaseTable targetDbTable = (DatabaseTable)targetDbObject;
+            if (sourceDbTable.Equals(targetDbTable))
+            {
+                throw new DataApiBuilderException(
+                message: $"Multiple-create for relationship: {relationshipName} at level: {nestingLevel} having the " +
+                $"source entity: {sourceEntityName} and target entity: {targetEntityName} backed by same database table is not yet supported.",
```

Comment:
$"Multiple-create for relationship: {relationshipName} at level: {nestingLevel} is not supported "+
$" because the source entity: {sourceEntityName} and target entity: {targetEntityName} are backed by same database table.",

nit: for clarity/conciseness